source/Halibut/Transport/SecureListener.cs

```diff
@@ -310,16 +315,32 @@ void SendFriendlyHtmlPage(Stream stream)
             // Disposing the StreamWriter will close the stream - it owns the stream
             using (var writer = new StreamWriter(stream, new UTF8Encoding(false)) { NewLine = "\r\n" })
             {
-                writer.WriteLine("HTTP/1.0 200 OK");
-                writer.WriteLine("Content-Type: text/html; charset=utf-8");
-                writer.WriteLine("Content-Length: " + message.Length);
-                foreach (var header in headers)
-                    writer.WriteLine($"{header.Key}: {header.Value}");
-                writer.WriteLine();
-                writer.WriteLine(message);
-                writer.WriteLine();
-                writer.Flush();
-                stream.Flush();
+                if (asyncHalibutFeature.IsEnabled())
+                {
+                    await writer.WriteLineAsync("HTTP/1.0 200 OK");
```

Comment:
nit: We could write to the stream directly (see sync and async version of [this](https://github.com/OctopusDeploy/Halibut/blob/93b2f5c0252c816d1586d4b8ae08e2c2b8472c38/source/Halibut/Transport/Protocol/MessageExchangeStream.cs#L90) ) if we wanted to be able to support cancellation